libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/AudioFocusDelegate.kt

```diff
@@ -9,11 +11,40 @@ interface AudioFocusDelegate {
      * Request audio focus. Send a request to obtain the audio focus
      * @return true on successful focus change request.
      */
+    @Deprecated(
+        "use #requestFocus(AudioFocusOwner, AudioFocusRequestCallback) instead",
+        ReplaceWith(
+            "requestFocus(AudioFocusOwner.) { }"
+        )
+    )
     fun requestFocus(): Boolean
 
+    /**
+     * Request audio focus. Send a request to obtain the audio focus
+     * @param owner specifies the owner for request
+     * @param callback invoked when a delegate processed audio request
+     */
+    fun requestFocus(owner: AudioFocusOwner, callback: AudioFocusRequestCallback) {
```

Comment:
Do we need to pass the `owner`? We are not using it 🤔

Should `owner` be passed to `requestFocus` 👇

Reply:
The implementation of this method is needed for backward compatibility only. The proper implementation will use the argument and will not call the deprecated version.

But it's only a source compatibility, binary is compatibility is broken here. Is it ok?

Reply:
Are you referring to https://github.com/mapbox/mapbox-navigation-android/pull/5652#discussion_r841613865? If so, how could we avoid that breaking change?

Reply:
The problem here is that technically the default implementation of this method is implicitly added to each class extending `AudioFocusDelegate` during the compilation (https://proandroiddev.com/kotlin-interface-default-implementation-how-does-it-work-3af5056e0c03). It can't be done ff such method is added after compilation of the derived class.